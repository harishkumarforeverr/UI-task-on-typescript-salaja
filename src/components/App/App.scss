@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,500;0,700;0,900;1,100;1,400;1,500;1,700;1,900&display=swap");

@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,400;1,500;1,700;1,900&display=swap");

.container {
  display: grid;
  grid-template-columns: 120px 1fr;
}
.head {
  width: 100%;
  height: 60px;

  background: #cc0000;
}
.lightRed {
  background-color: #951b30;
}
.sidebar {
  box-shadow: 0 6px 16px -8px #00000014, 0 9px 28px #0000000d,
    0 12px 48px 16px #00000008;
}
.boxshadow {
  box-shadow: 0 6px 16px -8px #00000014, 0 9px 28px #0000000d,
    0 12px 48px 16px #00000008;
}
body {
  height: 100vh;
  overflow: hidden;
  background-color: #f0f2f5;
  overflow-y: scroll;
}

* {
  font-family: Helvetica;
}
.appConatiner {
 
}
