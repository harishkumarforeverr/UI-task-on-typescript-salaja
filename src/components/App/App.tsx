import React from 'react';
import MainRoute from "../Route/MainRoute"




function App() {
  return (
    <>
    <MainRoute/>
    </>
  );
}

export default App;
