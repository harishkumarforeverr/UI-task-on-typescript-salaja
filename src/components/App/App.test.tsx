 import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import App from "./App"; 

test('renders learn react App', () => {
  const MyApp=()=>{
    return (
      <Router>
        <App />
      </Router>
    )
  }
  render(<MyApp />);
  const linkElement = screen.getByText("Home");
  expect(linkElement).toBeInTheDocument();
}); 

it("renders correctly", () => {
  const MyApp = () => {
    return (
      <Router>
        <App />
      </Router>
    );
  };
  const tree = renderer.create(<MyApp />).toJSON();
  expect(tree).toMatchSnapshot();
});
