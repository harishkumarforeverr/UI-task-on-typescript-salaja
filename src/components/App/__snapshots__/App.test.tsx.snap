// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    style={Object {}}
  >
    <div
      className="head"
    >
       
      <header
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed header_appbar mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header_toolbar css-hyum1k-MuiToolbar-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 header_typography css-ahj2mt-MuiTypography-root"
          >
            File
          </p>
          <p
            className="MuiTypography-root MuiTypography-body1 header_typography header_options_margin   css-ahj2mt-MuiTypography-root"
          >
            Options
          </p>
          <p
            className="MuiTypography-root MuiTypography-body1 header_typography header_options_margin   css-ahj2mt-MuiTypography-root"
          >
            Tools
          </p>
          <p
            className="MuiTypography-root MuiTypography-body1 header_typography header_options_margin   css-ahj2mt-MuiTypography-root"
          >
            Help
          </p>
        </div>
      </header>
    </div>
    <div
      className="container"
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "6rem 1fr",
          "height": "100%",
          "overflow": "hidden",
        }
      }
    >
      <div
        className="sidebar"
      >
         
        <div
          className="MuiDrawer-root MuiDrawer-docked MuiDrawer-paper  css-7ik032-MuiDrawer-docked"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
          >
            <div
              className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
            />
            <ul
              className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            >
              <a
                aria-current="page"
                className="MuiListItem-root MuiListItem-gutters MuiListItem-padding sidenav_listitem css-1p823my-MuiListItem-root active"
                disabled={false}
                href="/"
                onClick={[Function]}
              >
                <div
                  className="list_box selected-Tab-bgColor   MuiBox-root css-0"
                >
                  <img
                    alt="dd"
                    src="Images/home-Selected.svg"
                    style={
                      Object {
                        "fill": "red!important",
                      }
                    }
                  />
                </div>
                <p
                  className="MuiTypography-root MuiTypography-body1 menulist_typography selected-text-color  css-ahj2mt-MuiTypography-root"
                >
                  Home
                </p>
              </a>
              <a
                className="MuiListItem-root MuiListItem-gutters MuiListItem-padding sidenav_listitem css-1p823my-MuiListItem-root"
                disabled={false}
                href="/setting"
                onClick={[Function]}
              >
                <div
                  className="list_box false   MuiBox-root css-0"
                >
                  <img
                    alt="dd"
                    src="Images/setting-Unselected.svg"
                  />
                </div>
                <p
                  className="MuiTypography-root MuiTypography-body1 menulist_typography false  css-ahj2mt-MuiTypography-root"
                >
                  Setting
                </p>
              </a>
              <a
                className="MuiListItem-root MuiListItem-gutters MuiListItem-padding sidenav_listitem css-1p823my-MuiListItem-root"
                disabled={false}
                href="/quick_tuning"
                onClick={[Function]}
              >
                <div
                  className="list_box false   MuiBox-root css-0"
                >
                  <img
                    alt="dd"
                    src="Images/quick-Unselected.svg"
                  />
                </div>
                <p
                  className="MuiTypography-root MuiTypography-body1 menulist_typography_quickTuning false  css-ahj2mt-MuiTypography-root"
                >
                  Quick Tuning
                </p>
              </a>
              <a
                className="MuiListItem-root MuiListItem-gutters MuiListItem-padding sidenav_listitem sidenav_listitem_mpet css-1p823my-MuiListItem-root"
                disabled={false}
                href="/mpet"
                onClick={[Function]}
              >
                <div
                  className="list_box false   MuiBox-root css-0"
                >
                  <img
                    alt="dd"
                    src="Images/mpet-Unselected.svg"
                  />
                </div>
                <p
                  className="MuiTypography-root MuiTypography-body1 menulist_typography false  css-ahj2mt-MuiTypography-root"
                >
                  MPET
                </p>
              </a>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="body_container"
      >
        <div
          className="body_contain"
        >
          <div>
            <div
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "flexGrow": 1,
                  }
                }
              >
                <div
                  className="dashboard"
                >
                  <div
                    className="ConatinersIconsTitle"
                  >
                    <div>
                      <div
                        className="ant-card conatinerCard css-dev-only-do-not-override-1n7nwfa"
                        style={
                          Object {
                            "width": 300,
                          }
                        }
                      >
                        <div
                          className="ant-card-body"
                          style={Object {}}
                        >
                          <div>
                            <div
                              className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                            >
                              <label
                                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                                data-shrink={true}
                                id="selectlabelCommon"
                              >
                                Select Device*
                              </label>
                              <div
                                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-4gwozz-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                                onClick={[Function]}
                              >
                                <div
                                  aria-expanded="false"
                                  aria-haspopup="listbox"
                                  aria-labelledby="selectlabelCommon demo-simple-select"
                                  className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                                  id="demo-simple-select"
                                  onBlur={[Function]}
                                  onFocus={[Function]}
                                  onKeyDown={[Function]}
                                  onMouseDown={[Function]}
                                  role="button"
                                  tabIndex={0}
                                >
                                  Select Your Device
                                </div>
                                <input
                                  aria-hidden={true}
                                  className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                                  disabled={false}
                                  onAnimationStart={[Function]}
                                  onChange={[Function]}
                                  required={false}
                                  tabIndex={-1}
                                  value="Select Your Device"
                                />
                                <svg
                                  aria-hidden={true}
                                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                                  data-testid="ArrowDropDownIcon"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M7 10l5 5 5-5z"
                                  />
                                </svg>
                                <fieldset
                                  aria-hidden={true}
                                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                >
                                  <legend
                                    className="css-14lo706"
                                  >
                                    <span>
                                      Select Device*
                                    </span>
                                  </legend>
                                </fieldset>
                              </div>
                            </div>
                          </div>
                          <div>
                            <div
                              className="deatiles"
                            >
                              <span
                                className="ColorContainer"
                                style={
                                  Object {
                                    "background": "#cc0000",
                                  }
                                }
                              />
                              <span
                                className="numberText"
                              >
                                MCF8316EVM
                              </span>
                            </div>
                            <p
                              className="GuiSupport"
                            >
                              This GUI supports MCF8316EVM
                            </p>
                            <p
                              className="GuiSupport"
                              style={
                                Object {
                                  "fontSize": "12px",
                                  "opacity": "0.8",
                                }
                              }
                            />
                          </div>
                          <button
                            className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-default lightRed btn"
                            disabled={false}
                            onClick={[Function]}
                            style={
                              Object {
                                "opacity": "1",
                              }
                            }
                            type="button"
                          >
                            <span>
                              Select Device
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      className="cardsRigghtContainer"
                    >
                      <div
                        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": 300,
                          }
                        }
                      >
                        <div
                          className="ant-card-body"
                          style={Object {}}
                        >
                          <div
                            className="rightConatinesIcons"
                          >
                            <div
                              className="GetImages"
                            >
                              <img
                                className="ellipse_image"
                                src="Ellipse.png"
                              />
                              <div
                                className="GetImages_child"
                              >
                                <img
                                  alt="image_inner"
                                  className="image_inner"
                                  src="cardImage_1.png"
                                />
                                 
                              </div>
                            </div>
                          </div>
                          <div
                            className="titleRightConatiner"
                          >
                            Users's Guide
                          </div>
                        </div>
                      </div>
                      <div
                        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": 300,
                          }
                        }
                      >
                        <div
                          className="ant-card-body"
                          style={Object {}}
                        >
                          <div
                            className="rightConatinesIcons"
                          >
                            <div
                              className="GetImages"
                            >
                              <img
                                className="ellipse_image"
                                src="Ellipse.png"
                              />
                              <div
                                className="GetImages_child"
                              >
                                <img
                                  alt="image_inner"
                                  className="image_inner"
                                  src="cardImage_2.png"
                                />
                                 
                              </div>
                            </div>
                          </div>
                          <div
                            className="titleRightConatiner"
                          >
                            Product Page
                          </div>
                        </div>
                      </div>
                      <div
                        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": 300,
                          }
                        }
                      >
                        <div
                          className="ant-card-body"
                          style={Object {}}
                        >
                          <div
                            className="rightConatinesIcons"
                          >
                            <div
                              className="GetImages"
                            >
                              <img
                                className="ellipse_image"
                                src="Ellipse.png"
                              />
                              <div
                                className="GetImages_child"
                              >
                                <img
                                  alt="image_inner"
                                  className="image_inner"
                                  src="cardImage_3.png"
                                />
                                 
                              </div>
                            </div>
                          </div>
                          <div
                            className="titleRightConatiner"
                          >
                            Data Sheet
                          </div>
                        </div>
                      </div>
                      <div
                        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": 300,
                          }
                        }
                      >
                        <div
                          className="ant-card-body"
                          style={Object {}}
                        >
                          <div
                            className="rightConatinesIcons"
                          >
                            <div
                              className="GetImages"
                            >
                              <img
                                className="ellipse_image"
                                src="Ellipse.png"
                              />
                              <div
                                className="GetImages_child"
                              >
                                <img
                                  alt="image_inner"
                                  className="image_inner"
                                  src="cardImage_4.png"
                                />
                                 
                              </div>
                            </div>
                          </div>
                          <div
                            className="titleRightConatiner"
                          >
                            Tuning Page
                          </div>
                        </div>
                      </div>
                      <div
                        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": 300,
                          }
                        }
                      >
                        <div
                          className="ant-card-body"
                          style={Object {}}
                        >
                          <div
                            className="rightConatinesIcons"
                          >
                            <div
                              className="GetImages"
                            >
                              <img
                                className="ellipse_image"
                                src="Ellipse.png"
                              />
                              <div
                                className="GetImages_child"
                              >
                                <img
                                  alt="image_inner"
                                  className="image_inner"
                                  src="cardImage_5.png"
                                />
                                 
                              </div>
                            </div>
                          </div>
                          <div
                            className="titleRightConatiner"
                          >
                            Tools Page
                          </div>
                        </div>
                      </div>
                      <div
                        className="ant-card conatinerCardImage css-dev-only-do-not-override-1n7nwfa"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": 300,
                          }
                        }
                      >
                        <div
                          className="ant-card-body"
                          style={Object {}}
                        >
                          <div
                            className="rightConatinesIcons"
                          >
                            <div
                              className="GetImages"
                            >
                              <img
                                className="ellipse_image"
                                src="Ellipse.png"
                              />
                              <div
                                className="GetImages_child"
                              >
                                <img
                                  alt="image_inner"
                                  className="image_inner"
                                  src="cardImage_6.png"
                                />
                                 
                              </div>
                            </div>
                          </div>
                          <div
                            className="titleRightConatiner"
                          >
                            E2E testing
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="knowYourDevice"
                  >
                    <div
                      className="knowYourDevice_head"
                    >
                      <h1>
                        Know Your Device
                      </h1>
                    </div>
                    <div
                      className="knowYourDevice_context"
                    >
                      <div
                        className="imageKnowContainer"
                      >
                        <img
                          alt="Assests"
                          src="DashboardImageCircut.png"
                        />
                      </div>
                        
                      <div>
                        <p>
                          The MCF8316A is a 4.5-V to 35-V, 8-A peak integrated three-phase gate driver IC with sensorless field-oriented control for motor drive applications. It provides three accurately trimmed and temperature compensated halfbridge MOSFETS, gate drivers, charge pump, current sense amplifier, linear regulator for the external load and adjustable buck regulator. The I2C interface variant (MCF8316A) also provides a standard I2C interface for configuring the various device settings and reading fault diagnostic information through an external controller.
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  className="drawerConatiner_Parent"
                  style={
                    Object {
                      "marginLeft": "auto",
                    }
                  }
                >
                  <div
                    className="appConatiner_DashboardConatiner"
                  >
                    <div
                      className="appConatiner_drawer"
                      style={Object {}}
                    >
                      <div
                        className="i2cControl"
                        onClick={[Function]}
                      >
                        <span
                          aria-label="left"
                          className="anticon anticon-left rightIcon"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="left"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                            />
                          </svg>
                        </span>
                        <span
                          className="i2cLabel"
                        >
                          <p
                            className=""
                          >
                            I2C
                          </p>
                        </span>
                      </div>
                    </div>
                    <div
                      className="appConatiner_drawer faults_expanded"
                    >
                      <div
                        className="faultsControl"
                        onClick={[Function]}
                        style={Object {}}
                      >
                        <span
                          aria-label="left"
                          className="anticon anticon-left rightIcon"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="left"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                            />
                          </svg>
                        </span>
                        <span
                          className="faultsLabel"
                        >
                          <p
                            className=""
                          >
                            Faults
                          </p>
                          <div
                            className="smallCircle greenColor expandedGreen"
                          />
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
